### ISSUE TITLE
[Question] Parsing specific file

### ISSUE THREAD

**Issue by kristijanhusak** [nvim-treesitter/nvim-treesitter#1607]
Hi,

is there a way to parse specific files with tree-sitter and do queries on them, without opening them up in a buffer?

I want to figure out if I could leverage tree-sitter for [orgmode.nvim](https://github.com/kristijanhusak/orgmode.nvim) parsing, which requires parsing all the org files in specified glob path for purposes of agenda, searching, etc.

**Comment by theHamsta** [comment]
Yes, create a new language tree for your string (`:h languagetree.new()`)

**Comment by theHamsta** [comment]
Sorry, when you want to query you probably want to use (`:h Query:iter_matches()`) almost the whole tree-sitter API accepts a string instead of a buffer. So you could either just open background buffers or read your files to strings without opening them in Vim.

**Comment by kristijanhusak** [comment]
I'll give it a try, thanks.
